src/libs/utils.ts

```diff
@@ -139,15 +139,9 @@ export const log = (...args) => {
 
 const maybeCall = (maybeFn) => (_.isFunction(maybeFn) ? maybeFn() : maybeFn);
 
-/**
- * Takes any number of [predicate, value] pairs, followed by an optional default value.
- * Returns value() for the first truthy predicate, otherwise returns the default value().
- * Returns undefined if no predicate matches and there is no default value.
- *
- * DEPRECATED: If a value is not a function, it will be returned directly instead.
- * This behavior is deprecated, and will be removed in the future.
- */
-export const cond = (...args) => {
+type CondArgType<T> = [boolean, T | (() => T)];
+
+export const condTyped = <T>(...args: CondArgType<T>[]): T | undefined => {
   console.assert(
```

Comment:
feels like typescript should make this assertion unnecessary...  though something about the type signature above feels a little off...

Reply:
I agree, I should be able to move the assertion to the non-typed `cond` below.